daemon/logging/logging_outlets.go

```diff
@@ -170,5 +174,115 @@ func (o *SyslogOutlet) WriteEntry(entry logger.Entry) error {
 	default:
 		return o.writer.Err(s) // write as error as reaching this case is in fact an error
 	}
+}
+
+type FileOutlet struct {
+	file      *os.File
+	filename  string
+	formatter EntryFormatter
+	template  *template.Template
+	writer    io.Writer
+}
+
+func (self *FileOutlet) WriteEntry(entry logger.Entry) error {
+	bytes, err := self.formatter.Format(&entry)
+	if err != nil {
+		return err
+	}
+
+	if err := self.reOpenIfNotExists(); err != nil {
+		return err
+	}
```

Comment:
> That's quite a lot of overhead.
Actually no. In real life there is no overhead at all. zrepl doesn't log hundred thousands messages per second.  And if, I repeat if, for somebody it'll have some overhead, somebody can just don't use it and use syslog or tcp instead.

> I guess because you're assuming there's external rotation?
Correct.

> we should include log rotation functionality
I don't think that's good idea. zrepl is for snapshots and newloslogd for log rotation. Every tool for single task.